### DISCUSSION TITLE
'use client' In the parent component...

### DISCUSSION THREAD

**Discussion by SomeOne-49** [vercel/next.js#60330] [App Router] [answered]
If I use 'use client' in the parent component (<Navbar/> for example) will all of its children be client-side and I won't be able to take advantage of the server-side features within them?!

**Comment by icyJoseph** [comment, marked as answer]
Depends... if your parent component owns the children it renders. For example, this extremely simplified case:

```jsx
'use client'

import { ThisComponent } from 'somewhere';

export const Parent = () => {
  return <ThisComponent />
}
```

> Let's assume that `ThisComponent` did not use any client hooks (useState, useReducer, etc).

Because it is imported into a client module, it will also be a client module.


Versus doing something like:

```jsx
'use client'


export const Parent = ({children} ) => {
  return <>{children}</>
}
```

And using it in yet another Server Component as:

```jsx
// app/page.tsx for example

import { Parent } from 'somewhere-else';
import { ThisComponent } from 'somewhere';

export default function Page() {
   return (
      <Parent>
         <ThisComponent />
      </Parent>
   )
}
```

This is the composition pattern at play. This way `ThisComponent` does not become a client module.
